lib/taskana-core/src/test/java/acceptance/task/transfer/TransferTaskAccTest.java

```diff
@@ -435,4 +405,224 @@ void should_NotSetTheTransferFlagWithinBulkTransfer_When_SetTransferFlagNotReque
+  }
+
+  @WithAccessId(user = "teamlead-1", groups = GROUP_1_DN)
+  @Test
+  void should_BulkTransferOnlyValidTasksAndSetOwner_When_SomeTasksToTransferCauseExceptions()
+      throws Exception {
+    final Workbasket destinationWb =
+        taskanaEngine
+            .getWorkbasketService()
+            .getWorkbasket("WBI:100000000000000000000000000000000009");
+    final Instant before = Instant.now().truncatedTo(ChronoUnit.MILLIS);
+    List<String> taskIdList =
+        Arrays.asList(
+            "TKI:000000000000000000000000000000000007", // working
+            "TKI:000000000000000000000000000000000041", // NotAuthorized READ
+            "TKI:000000000000000000000000000000000041", // NotAuthorized READ
+            "TKI:200000000000000000000000000000000008", // NotAuthorized TRANSFER
+            "", // InvalidArgument
+            null, // InvalidArgument
+            "TKI:000000000000000000000000000000000099", // not existing
+            "TKI:100000000000000000000000000000000006"); // already completed
+
+    BulkOperationResults<String, TaskanaException> results =
+        taskService.transferTasksWithOwner(
+            "WBI:100000000000000000000000000000000009", taskIdList, "teamlead-1");
+
+    // check for exceptions in bulk
+    assertThat(results.containsErrors()).isTrue();
+    assertThat(results.getErrorMap().values()).hasSize(6);
+    assertThat(results.getErrorForId("TKI:000000000000000000000000000000000041").getClass())
+        .isEqualTo(NotAuthorizedOnWorkbasketException.class);
+    assertThat(results.getErrorForId("TKI:200000000000000000000000000000000008").getClass())
+        .isEqualTo(NotAuthorizedOnWorkbasketException.class);
+    assertThat(results.getErrorForId("TKI:000000000000000000000000000000000099").getClass())
+        .isEqualTo(TaskNotFoundException.class);
+    assertThat(results.getErrorForId("TKI:100000000000000000000000000000000006").getClass())
+        .isEqualTo(InvalidTaskStateException.class);
+    assertThat(results.getErrorForId("").getClass()).isEqualTo(TaskNotFoundException.class);
+    assertThat(results.getErrorForId(null).getClass()).isEqualTo(TaskNotFoundException.class);
+
+    Task transferredTask = taskService.getTask("TKI:000000000000000000000000000000000007");
+    assertTaskIsTransferred(
+        transferredTask.asSummary(), destinationWb, true, TaskState.READY, before, "teamlead-1");
+
+    Task notTransferredTask = taskService.getTask("TKI:200000000000000000000000000000000008");
+    assertThat(notTransferredTask).isNotNull();
+    assertThat(notTransferredTask.isTransferred()).isFalse();
+    assertThat(notTransferredTask.getWorkbasketKey()).isEqualTo("TPK_VIP");
+
+    notTransferredTask = taskService.getTask("TKI:100000000000000000000000000000000006");
+    assertThat(notTransferredTask).isNotNull();
+    assertThat(notTransferredTask.isTransferred()).isFalse();
+    assertThat(notTransferredTask.getWorkbasketKey()).isEqualTo("TEAMLEAD-1");
```

Comment:
Why don't you check "TKI:000000000000000000000000000000000041" here?

Reply:
Yes I will check it. But I need to use runAsAdmin() to run taskService.getTask("TKI:000000000000000000000000000000000041"), since the current user does not have READ permission